src/decorators/validated-string-api-property.decorator.ts

```diff
@@ -1,25 +1,30 @@
 import { applyDecorators } from '@nestjs/common';
 import { ApiProperty } from '@nestjs/swagger';
-import { IsOptional, Length } from 'class-validator';
+import { IsOptional, Length, Matches } from 'class-validator';
 
 interface Options {
   description: string;
-  minLength: number;
-  maxLength: number;
-  example?: string;
+  length?: number;
+  minLength?: number;
+  maxLength?: number;
   required?: boolean;
+  pattern?: RegExp;
+  example?: string;
   default?: string;
 }
 
-export const ValidatedStringApiProperty = ({ description, minLength, maxLength, example, required, default: theDefault }: Options) => {
+export const ValidatedStringApiProperty = ({ description, length, minLength, maxLength, required, pattern, example, default: theDefault }: Options) => {
+  minLength = minLength ?? length;
+  maxLength = maxLength ?? length;
```

Comment:
I've changed these lines to 
```
minLength = length ?? minLength;
maxLength = length ?? maxLength;
```

Reply:
Also, to prevent this unintended use, could we do something like:
```
type Options =
  | {
      description: string;
      length: number;
      minLength?: undefined;
      maxLength?: undefined;
      required?: boolean;
      pattern?: RegExp;
      example?: string;
      default?: string;
    }
  | {
      description: string;
      length?: undefined;
      minLength: number;
      maxLength: number;
      required?: boolean;
      pattern?: RegExp;
      example?: string;
      default?: string;
    };
```

This way, the developer must either specify `length` OR (`minLength` and `maxLength`)

Reply:
Ah apologies, I've just rebased onto main and seen that a function has been added that throws errors if both length and (minLength or maxLength) are specified

Reply:
> Also, to prevent this unintended use, could we do something like...

That is an interesting idea - I think if it was a type where we thought "length v.s. min + max length" was the only variation of this type then it might be good to do, but I think this type is likely to grow to have a few of those options and it'll become very hard to manage.

So I'm happy to leave it without that type change